### PR TITLE
feat(components): watermark component

### PR DESCRIPTION
**Pull request by TomatoDroid**
be able to set text,multi-text,image as watermark

Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### <samp>🤖 Generated by Copilot at f245ac4</samp>

This pull request adds a new Watermark component to the element-plus package, which allows users to add a watermark effect to any element with text or image content. The pull request includes the component code, types, hooks, utils, and tests, as well as the documentation and examples for the component. The component is imported and exported in the `component.ts` file and the documentation sidebar is updated in the `component.json` file.

## Related Issue

Fixes #\_\_\_.

### <samp>🤖 Generated by Copilot at f245ac4</samp>

*  Add a new Watermark component to the element-plus package, which allows users to add a watermark effect to any element ([link](https://github.com/element-plus/element-plus/pull/14236/files?diff=unified&w=0#diff-7a9b4ae26499c9b0758fdbc9129f3a70a8b3a2a08003f09106526dfc3b558d22R1-R75), [link](https://github.com/element-plus/element-plus/pull/14236/files?diff=unified&w=0#diff-5bd1d4ae0fe297de0e867f25fb4d80e6b5d711e8b6e2aee580fc1b781acd870eR1-R5), [link](https://github.com/element-plus/element-plus/pull/14236/files?diff=unified&w=0#diff-4c2b9c65a56232995f1a6a6b3029ab0f50efea84d0fd5eebfc0828d5d767637dR1-R103), [link](https://github.com/element-plus/element-plus/pull/14236/files?diff=unified&w=0#diff-3578c6098a38c06808d72d2f5ce62e75a9b0e5a214d97a7e426d2bc275f09554R1-R5), [link](https://github.com/element-plus/element-plus/pull/14236/files?diff=unified&w=0#diff-b9fc2a4052a71a87e3419e43ec2ba2a4817b3c9a864569806fdd4915e5505ec8R1-R5), [link](https://github.com/element-plus/element-plus/pull/14236/files?diff=unified&w=0#diff-e360685d73a174d62eb4f65c7e77eeed0b72ca5b7db62d4ddfcfe1021205443aR1-R7), [link](https://github.com/element-plus/element-plus/pull/14236/files?diff=unified&w=0#diff-36fb9d011154596a2ee915203af4e42826711c178efcd7761e99dd8eab50cdfeR1-R145), [link](https://github.com/element-plus/element-plus/pull/14236/files?diff=unified&w=0#diff-dcfc161cd2c74f2d7ac2a0ed0ea4c220cc84d7faa4816f6f73e56d47a760f2c5R1-R37), [link](https://github.com/element-plus/element-plus/pull/14236/files?diff=unified&w=0#diff-bb55dba6ad270e7709671cf8999d9338e5e161b4afd195e0f1e35046891160abR1-R70), [link](https://github.com/element-plus/element-plus/pull/14236/files?diff=unified&w=0#diff-c6c90c95997cf2bd5412dcad208a28a43a70addb4960c01a4a22440228b0a315R1-R225), [link](https://github.com/element-plus/element-plus/pull/14236/files?diff=unified&w=0#diff-360f079f95aa081261f14bddac98bfbd9b59deeda273cb1b9fc15b7eaf11a2f9R105), [link](https://github.com/element-plus/element-plus/pull/14236/files?diff=unified&w=0#diff-360f079f95aa081261f14bddac98bfbd9b59deeda273cb1b9fc15b7eaf11a2f9R208))
*  Add a link and text for the Watermark component in the `component.json` file, which generates the documentation sidebar for the component library ([link](https://github.com/element-plus/element-plus/pull/14236/files?diff=unified&w=0#diff-aa195b9bfa86f84651be846117d76b4cf1c6f902a91cc08a0f043c10b14d90a0R326-R329))
*  Add a unit test for the Watermark component using vite and vue-test-utils, which tests the component's creation and slots ([link](https://github.com/element-plus/element-plus/pull/14236/files?diff=unified&w=0#diff-7d4de8352ca1cb631c339681c0eee99e9c2f85ad0ab606cbb633c3c94aa4e9eaR1-R22))

### CODE REVIEW

**Comment by btea** [comment]
You shouldn't commit the `pnpm-lock.yaml` file.

**Comment by TomatoDroid** [comment]
> You shouldn't commit the `pnpm-lock.yaml` file.

I solved it。

**Review by ryuhangyeong — COMMENTED** [review]
@btea 

Do we accept new components?

This is a good thing, but we need some design guide (ex. figma)

**Comment by btea** [comment]
I agree, this component looks great. What do you think? cc @element-plus/backers

**Comment by kooriookami** [comment]
This a nice component, we should accept it after reviewed.

File: docs/examples/watermark/image.vue [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<template>
+  <el-watermark :width="50" :height="50" image="/apple-touch-icon.png">
```
Comment (iamkun) [inline docs/examples/watermark/image.vue:2]: it is better to use a SVG image like "https://element-plus.org/images/element-plus-logo-small.svg" , to make it compatible in dark mode

**Review by YunYouJun — APPROVED** [review]
The dynamic example is well done.

File: docs/examples/watermark/image.vue (hunk 1 above)
Comment (TomatoDroid) [reply docs/examples/watermark/image.vue]: Thanks for the advice,I finished it.
image

**Review by kooriookami — APPROVED** [review]

**Pull request merged**